File: BackEnd/app/Models/Material.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Material extends Model
{
    use HasFactory;

    /**
     * MODIFIKASI PENTING:
     * Karena di file .env Anda hanya ada satu database (db_spectaacademy),
     * kita harus menonaktifkan baris connection di bawah ini.
     * Jika diaktifkan, Laravel akan error karena mencari database 'pgsql_materi' yang tidak ada di .env.
     */
    // protected $connection = 'pgsql_materi'; 

    protected $table = 'materials';
    protected $primaryKey = 'material_id';
    public $incrementing = true;
    protected $keyType = 'int';

    protected $fillable = [
        'class_id', 
        'user_id', 
        'title', 
        'material_name', 
        'file_path', 
        'week'
    ];

    /**
     * Relasi ke Model ClassModel
     */
    public function classModel()
    {
        return $this->belongsTo(ClassModel::class, 'class_id', 'class_id');
    }

    /**
     * Relasi ke Pengajar (User)
     */
    public function pengajar()
    {
        return $this->belongsTo(User::class, 'user_id', 'usersID');
    }
}

File: Services/materi-service/internal/models/material.go
package models

import (
	"time"
)

type Material struct {
	MaterialID   uint      `gorm:"primaryKey;column:material_id" json:"material_id"`
	ClassID      int       `gorm:"column:class_id" json:"class_id"`
	UserID       int       `gorm:"column:user_id" json:"user_id"`        // ✅ Ubah dari sql.NullInt64 ke int
	SubjectName  string    `gorm:"column:material_name" json:"subject_name"`
	Title        string    `gorm:"column:title" json:"title"`
	Week         int       `gorm:"column:week;default:1" json:"week"`
	FilePath     string    `gorm:"column:file_path" json:"file_path"`
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}